### PR TITLE
PEP 8 -- Style Guide for Python Code

### PR DESCRIPTION
**Pull request by clementval**
As we probably will have some new code from Fabian in this repo sometime soon and we are talking about C++ code style for his part, I thought will be also nice to clean up our Python code and maybe later C and Fortran to have an homogeneous repository. 

So I made the change on pp_ser.py to follow the PEP 8 style guide. I'm not an hardcore Python programmer so I'm not sure if it makes total sense but as we have only few Python code I think it's pretty straight forward to follow this. There is plenty of "checker" on the Internet and most IDE will check this style for you (e.g. PyCharm). If this is fine, I'll apply this style also to the other Python files (Serializer/Visualizer).

### CODE REVIEW

File: python/pp_ser.py [hunk 1]
```diff
@@ -327,7 +328,8 @@ def __ser_register(self, args):
         datatypes = dict(integer=["'int'", 'ppser_intlength'], real=['ppser_realtype', 'ppser_reallength'])
 
         if dirs[1] not in datatypes:
-            self.__exit_error(directive = args[0], msg = 'Data type "{0}" not understood. Valid types are: {1}'.format(dirs[1], ', '.join('"' + d + '"' for d in datatypes.keys())))
+            self.__exit_error(directive=args[0], msg='Data type "{0}" not understood. Valid types are: {1}'.
+                              format(dirs[1], ', '.join('"' + d + '"' for d in datatypes.keys())))
```
Comment (andyspiros) [inline python/pp_ser.py:332]: Does PEP8 specify that in these cases one can put a newline just after the dot? I always wonder where should I break long `.format` statements...

**Comment by andyspiros** [comment]
My impression is, this is a very good step in the right direction. I still see some room for improvement, like we should probably avoid property and function names starting with double underscore. Not to mention I have a very personal feeling ppser should be redesigned, but this is just a feeling.

I think there is a built-in PEP8 checker somewhere (I have one in vim but I never remember how to use it). What does it say?

File: python/pp_ser.py [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 from __future__ import print_function
-import linecache
 
+import filecmp
+import linecache
+import os
+import re
+import shutil
+import sys
+import tempfile
```
Comment (andyspiros) [inline python/pp_ser.py:13]: How about importing multiple modules in the same line? I usually load standard modules like `os`, `sys`, `re`, `shutil` together.

File: python/pp_ser.py [hunk 3]
```diff
@@ -27,100 +33,97 @@
...
+__license__ = 'GPL'
+__version__ = '0.1'
+__date__ = 'Sun Mar 23 22:06:44 2014'
+__email__ = 'oliver.fuhrer@meteoswiss.ch'
 
-# modules
-import sys, os, tempfile, re, filecmp, shutil
 
-def toASCII(text):
+def to_ascii(text):
```
Comment (andyspiros) [inline python/pp_ser.py:44]: This is a function I added to support both versions of Python. Do you know if there is a better way to achieve this? Also, do you know if ppser works correctly with both?

**Review by andyspiros — APPROVED** [review]
I like this PR :+1:

File: python/pp_ser.py (hunk 2 above)
Comment (clementval) [reply python/pp_ser.py:13]: Well it is a recommendation of PEP 8 to not have multiple module import on the same line. Then when can accommodate the PEP 8 style to what we want

Comment (andyspiros) [reply python/pp_ser.py:13]: I was just curious. It means I break PEP8 whenever I use my multiple imports. I'm proud of this now.

File: python/pp_ser.py (hunk 3 above)
Comment (clementval) [reply python/pp_ser.py:44]: Well this is also something lacking we should have some unit tests for this ... at least have a few fortran inputs and parse them as regression tests.

**Comment by pspoerri** [comment]
Nice 👍

File: python/pp_ser.py (hunk 1 above)
Comment (clementval) [reply python/pp_ser.py:332]: should break after the operator.

**Comment by clementval** [comment]
@andyspiros well I didn't find any built-in tool but there is this package 
https://pypi.python.org/pypi/pep8

**Review by pspoerri — APPROVED** [review]
Very nice!

**Comment by clementval** [comment]
@andyspiros I agree with you about the "room for improvement" and the "redesigned". If times permits, I'll make for suggestions in the future.

**Pull request merged**